examples/sat/test_schoening.py

```diff
@@ -30,7 +34,34 @@ def test_solve(rng) -> None:
         # check stats
         assert tracker.get_stats(schoening_with_randomness) == QueryStats(
             classical_actual_queries=2,
-            classical_expected_queries=pytest.approx(2.0009153259895374),
-            quantum_expected_classical_queries=pytest.approx(2.0009153318077804),
+            classical_expected_queries=pytest.approx(64),
+            quantum_expected_classical_queries=pytest.approx(3.3703703703703702),
```

Comment:
The values have changed from the previous implementation; how do we verify any of the statistics? Especially interesting since e.g. classical_expected_queries changed drastically...

Reply:
Yeah, we need to pick $O(\log_{1-f}(\epsilon))$ samples, where $f$ is probability of sampling a solution and $\epsilon$ is the (upperbound on) required failure probability.
The method basically samples 3 times that, and can technically fail (i.e. return None even if a solution exists).

Reply:
tracking this in https://github.com/qubrabench/qubrabench/issues/106 (can think about it after this PR)

Reply:
looks like this is the current performance bottleneck